auditlog/audit_log.go

```diff
@@ -0,0 +1,227 @@
+package auditlog
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"reflect"
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/closeable"
+	"github.com/fabric8-services/fabric8-common/convert"
+	"github.com/fabric8-services/fabric8-common/errors"
+	"github.com/fabric8-services/fabric8-common/log"
+
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+// AuditLog an audit log record to track usage of the service
+type AuditLog struct {
+	ID          uuid.UUID   `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"`
+	CreatedAt   time.Time   `json:"created_at,omitempty"`
+	IdentityID  uuid.UUID   `sql:"type:uuid"` // TODO: or should we store the sub claim instead as a string: no need to convert it into an UUID each time
```

Comment:
It's fine to keep it uuid since that's the format everywhere.

Reply:
ok, that's what this PR uses anyways. I'll remove the comment, then.